app/src/main/java/org/groundplatform/android/ui/settings/LocaleHelper.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.groundplatform.android.ui.settings
+
+import android.content.Context
+import android.content.res.Configuration
+import androidx.preference.PreferenceManager
+import java.util.Locale
+import org.groundplatform.android.Config.DEFAULT_LANGUAGE
+
+fun onAttach(context: Context, defaultLanguage: String = DEFAULT_LANGUAGE): Context {
+  val lang = getPersistedLanguage(context, defaultLanguage)
+  return setLocale(context, lang)
+}
+
+private fun getPersistedLanguage(context: Context, defaultLanguage: String): String {
+  val preferences = PreferenceManager.getDefaultSharedPreferences(context)
+  return preferences.getString(Keys.LANGUAGE, defaultLanguage) ?: DEFAULT_LANGUAGE
```

Comment:
Let's also not hide the "default" behavior in the persistence layer. Note that the "default" should be the system language. `en` should only be the fallback when the system language is not supported.

Reply:
Is this approach looks correct?


```kt
fun getAppLanguage(): String {
    preferences.getString(Keys.LANGUAGE, null)?.let { return it }
    return locale.language ?: DEFAULT_LANGUAGE
  }
```